plugins/database_api/api.cpp

```diff
@@ -213,14 +213,9 @@ optional<signed_block> plugin::api_impl::get_block(uint32_t block_num) const {
 //////////////////////////////////////////////////////////////////////
 
 DEFINE_API(plugin, set_block_applied_callback) {
-    CHECK_ARG_SIZE(1);
-    block_applied_callback_result_type type = block;
-    auto arg = args.args->at(0);
-    try {
-        type = arg.as<block_applied_callback_result_type>();
-    } catch (...) {
-        ilog("Bad argument (${a}) passed to set_block_applied_callback, using default", ("a",arg));
-    }
+    PLUGIN_API_VALIDATE_ARGS(
+        (block_applied_callback_result_type, type)
```

Comment:
`set_block_applied_callback` reads parameters differently from other api methods. It tries to cast 1st argument to `block_applied_callback_result_type` and sets default value if cast failed. This is to preserve previous HF behaviour, where 1st argument can be any integer.
`PLUGIN_API_VALIDATE_ARGS` can't do this.
So we can either break backward compatibility completely forcing argument to be `block_applied_callback_result_type` or use old code (somewhat inconsistent), or modify macro for this case (not easy). @afalaleev, @s-medvedev, what do you think?

btw, we can make 1st argument optional

Reply:
Ok, lets don't use `PLUGIN_API_VALIDATE_ARGS` and write a custom argument parsing for this API method. I think it will be better.

Reply:
Fixed